src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -180,6 +210,42 @@ private Node newFloatingCommentFromBuffer() {
       return c;
     }
 
+    /**
+     * If the parent node can survive code generation with an extra EMPTY node child, then outputs
+     * floating comments by attaching to a new EMPTY node before the current node. If the parent is
+     * sensitive about its children count/order, then try to put the comment in current node. If the
+     * current node doesn't accept comments (for example a BLOCK node will ignore any comments
+     * mapped to it), then attach the comments to the parent node. The parent node might not be the
+     * best place to hold the comment but at least we don't lose the comment and we don't break code
+     * generation.
+     *
+     * @param n
+     */
+    private void outputFloatingCommentFromBuffer(Node n) {
+      Node parent = n.getParent();
+      if (TOKENS_SET_ACCEPT_EMPTY_CHILD.contains(parent.getToken())) {
+        parent.addChildBefore(newFloatingCommentFromBuffer(), n);
+      } else if (canHaveComment(n)) {
+        linkCommentBufferToNode(n);
+      } else {
+        outputFloatingCommentFromBuffer(parent);
+      }
+    }
+
+    private boolean canHaveComment(Node n) {
+      if (TOKENS_SET_IGNORE_COMMENTS.contains(n.getToken())) {
+        return false;
+      }
+
+      // Special case: GETPROP node's second child(the property name) loses comments.
```

Comment:
GETPROP has only two children so second is last.
Changed to getSecondChild() so it's consistent.